packages/bpk-component-graphic-promotion/src/BpkGraphicPromo.tsx

```diff
@@ -0,0 +1,197 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { cssModules } from 'bpk-react-utils';
+import BpkText from 'bpk-component-text';
+import BpkButton from 'bpk-component-button';
+
+import STYLES from './BpkGraphicPromo.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+const ACCESSIBILITY_KEYS = {
+  Enter: 13 /* Enter */,
+  Space: 32 /* Space */,
+};
+
+const isAccessibilityClick = (event: React.KeyboardEvent<HTMLElement>) =>
+  Object.keys(ACCESSIBILITY_KEYS).includes(event.key) ||
+  Object.values(ACCESSIBILITY_KEYS).includes(event.keyCode || event.which);
+
+export const TEXT_ALIGN = {
+  start: 'start',
+  center: 'center',
+  end: 'end',
+} as const;
+
+export type Props = {
+  buttonText: string;
+  className?: string | null;
+  contentId?: string | null;
+  headline: string;
+  invertVertically: boolean;
+  onClick: () => void;
+  sponsor?: {
+    label: string;
+    logo: string;
+    altText: string;
+  } | null;
+  style?: {};
+  subheading?: string | null;
+  tagline?: string | null;
+  textAlign: typeof TEXT_ALIGN[keyof typeof TEXT_ALIGN];
+};
+
+const constructAriaLabel = (
+  buttonText: Props['buttonText'],
+  headline: Props['headline'],
+  sponsor: Props['sponsor'],
+  subheading: Props['subheading'],
+  tagline: Props['tagline'],
+) => {
+  const text: string[] = [];
+  const addText = (value: string) => value && text.push(value);
+
+  if (sponsor) {
+    addText(sponsor.label);
+    addText(sponsor.altText);
+  } else if (tagline) {
+    addText(tagline);
+  }
+
+  if (subheading) {
+    addText(subheading);
+  }
+
+  addText(headline);
+  addText(buttonText);
+
+  return text.join('. ');
+};
+
+const BpkGraphicPromo = ({
+  buttonText,
+  className = null,
```

Comment:
Stop using soon-to-be-deprecated `.defaultProps` and start using the recommended default arguments.

https://github.com/reactjs/rfcs/blob/createlement-rfc/text/0000-create-element-changes.md#deprecate-defaultprops-on-function-components